src/Service/EnterpriseSearchService.php

```diff
@@ -471,7 +478,22 @@ private function fetchEngines(): array
             throw new IndexingServiceException('Invalid response format for listEngines; missing "results"');
         }
 
-        $results = $response['results'] ?? [];
+        return $response;
+    }
+
+    private function fetchEngines(): array
+    {
+        $response = $this->fetchPaginatedEngines(1);
+
+        $results = $response['results'];
```

Comment:
```suggestion
        $results = $response['results'] ?? [];
```

Reply:
The function errors if this doesn't exist so this would be redundant wouldn't it?

```
if (!array_key_exists('results', $response)) {
    throw new IndexingServiceException('Invalid response format for listEngines; missing "results"');
}
```

Reply:
The check if for the existence of the key in the array, not of its value, so if it existed and were null, it would still get through the check and then fail when trying to merge the arrays. It could actually be `?: []` to also cater for any false-y values. But probably not critical.

Reply:
Ok, I've included an `is_array` check for 'results' as according to the api docs it will either be defined with an array of results or not included at all if there are no results. This will handle this situation.